core/lib/queued_job_processor/src/lib.rs

```diff
@@ -46,6 +46,7 @@ pub trait JobProcessor: Sync + Send {
     /// Function that processes a job
     async fn process_job(
         &self,
+        job_id: &Self::JobId,
```

Comment:
The only reason why I need `job_id` to be inside `process_job` is because for `prover-fri-compressor` the `job_id`represents the L1 batch number. I *can* remove it from here and instead introduce a more generic span inside `run` that has `job_id` as one of the attributes. But then it wouldn't be clear that for some spans that `job_id` is L1 batch number. Not sure if you would be okay with that.